packages/in_app_purchase/in_app_purchase_storekit/pigeons/messages.dart

```diff
@@ -238,7 +238,7 @@ abstract class InAppPurchaseAPI {
   SKProductsResponseMessage startProductRequest(
       List<String> productIdentifiers);
 
-  void finishTransaction(Map<String, String?> finishMap);
+  void finishTransaction(Map<String, Object?> finishMap);
```

Comment:
Even though the intended type here should be a nullable String, when pigeon generates obj C code, it is generated as type NSString. NSString is nullable in objective c, but when Xcode generates the corresponding swift method signature, it converts into a String, not a String?. NSString also cannot be annoted with _Nullable as it is a generic.

Reply:
im confused. why is NSString generic?

Reply:
https://github.com/flutter/packages/pull/6561#discussion_r1571294947

Reply:
Sorry, its not generic, but you can't specify a type inside a NSDictionary to be nullable. (i dont think)
Specifically here -
https://github.com/flutter/packages/blob/74de10af17274ab3d93a264b215ba19ec20e7ba6/packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/messages.g.h#L268
What I really want is for this line in swift 
https://github.com/flutter/packages/blob/74de10af17274ab3d93a264b215ba19ec20e7ba6/packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift#L227
To be `finishMap: [String: String?]`. But the automatic conversion turns all NSStrings as `String` and not `String?`. And because I can't specify the NSString as extra nullable, I have to make it `Object?` (in pigeon) so it gets generated into `id` in objc, which then will turn into `Any`, and thus allow for nullability
Not sure if that makes sense lol

Reply:
Using `Any` to represent an optional can be error-prone. 
Why do you want `[String:String?]`? Is there any issue with using `[String:String]`?

Reply:
because the key (in this case `transactionidentifier`) is never nil, but there are cases where the value will be nil. but that is still an acceptable form, since it signals a cancelled transaction

Reply:
Can you point me to the place where it's used and where it distinguishes the cancelled transaction vs no transaction?

Reply:
https://github.com/flutter/packages/blob/74de10af17274ab3d93a264b215ba19ec20e7ba6/packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift#L237

Reply:
are you referring to the comments below the code?

Reply:
Not sure how `pendingTransactions` is related to the dictionary type. Can you explain how using `[String:String]` type would fail to work here?

Reply:
Yea I meant the comments
Im not understanding the question i think? finishMap should be `[String:String?]` because `finishMap["transactionIdentifier"]` should be allowed to be null.

Reply:
If it's just `[String:String]` then `finishMap["transactionIdentifier"] ` wouldn't be allowed to be null?

Reply:
https://github.com/flutter/packages/blob/871a24b977c5e774cf9fc6a8c9f03d3338313125/packages/in_app_purchase/in_app_purchase_storekit/lib/src/store_kit_wrappers/sk_payment_transaction_wrappers.dart#L218

You can see on the dart side that the finishMap arg it gets called with has [String:String?] as well - I genuinely do not think it would be correct to make it [String:String]. Once this gets converted to using Swift pigeon, I can change it from `String:Any` to `String:String?`, so this should be temporary

Reply:
> finishMap should be [String:String?] because finishMap["transactionIdentifier"] should be allowed to be null.
> If it's just [String:String] then finishMap["transactionIdentifier"]  wouldn't be allowed to be null, right?

Not really. `dict[key]` always returns optional. If type of `finishMap` is `[String:String]`, then the type of `finishMap["transactionIdentifier"]` would be `String?`. 

If you use `[String:String?]` as the type, then `finishMap["transactionIdentifier"]` would be `String??` (double optional).